packages/preset-uno/src/variants/index.ts

```diff
@@ -35,7 +35,8 @@ export const variantNegative: Variant = {
         matcher: matcher.slice(1),
         body: (body) => {
           body.forEach((v) => {
-            if (v[1]?.toString().match(/^\d/))
+            v[1] = v[1]?.toString().replace(/[0-9.]+[a-z]+/, i => `-${i}`)
+            if (!v[0].includes('reverse') && v[1]?.toString().match(/^\d/))
```

Comment:
```suggestion
            if (!v[0].includes('reverse') && v[1]?.toString().match(/^\d/))
```

`v[0].includes('reverse')` this one feels a bit tricky to me. Can we have a better workaround?

Reply:
Actually we can remove L39-40, and all tests still passed.

Reply:
Sure, let's do that then